README.md

```diff
@@ -78,103 +101,366 @@ Modify the app's **android/app/src/main/AndroidManifest.xml** to include the fol
 
 ---
 
-# Importing and Instantiation
+# API Guide
+
+## Important Information 
+The client must be instantiated and initialized before using the rest of the API. Failing to do so
+will result in a lack of functionality (this will be logged out to the system - this means that
+during testing it will appear where console.logs appear).
+
+## Raygun Specific Types
+This segment outlines the type declarations for some Raygun4ReactNative specific objects. Each of
+these object types can be imported into your program and used (as the examples throughout the
+rest of the API guide show).
 
-Import the RaygunClient using:
+<br/>
+
+###  RaygunClientOptions 
+The RaygunClientOptions type gets used to parse information into the RaygunClient during the [init](#initraygunclientoptions)
+function. Each field within the object is optional, however, failing to declare a field will result
+in the option defaulting to its type specific default value (except for custom Endpoints
+which will default to the Raygun application Endpoint). 
 
 ```typescript
-import RaygunClient from 'raygun4reactnative';
+export type RaygunClientOptions = {
+  apiKey?: string; // Found in the 'Application Settings' within the Raygun application.
+  version?: string; // The version of YOUR application.
+  enableCrashReporting?: boolean; // Enables the use of Crash Reporting.
+  disableNativeCrashReporting?: boolean; // If Crash Reporting is enabled, this disables reporting on native crashes.
+  enableRealUserMonitoring?: boolean; // Enables Real User Monitoring.
+  disableNetworkMonitoring?: boolean; // If Real User Monitoring is enabled, this disables monitoring network usage.
+  customCrashReportingEndpoint?: string; // Some alternate URL to send crash reports to.
+  customRealUserMonitoringEndpoint?: string; // Some alternate URL to send real user monitoring data to.
+  onBeforeSendingCrashReport?: BeforeSendHandler; // A BeforeSendHandler function (see below).
+  ignoredURLs?: string[]; // An Array of URLs to ignore when network monitoring.
+};
 ```
 
-Initialize the RaygunClient using the `init(RaygunClientOption)` function.
+<br/>
+<br/>
 
+###  BeforeSendHandler 
+The BeforeSendHandler acts as an intermediate function between your application and Raygun. This
+function gets parsed a [CrashReportPayload](#crashreportpayload) and returns a [CrashReportPayload](#crashreportpayload). 
+Before the [CrashReportPayload](#crashreportpayload) is sent to Raygun, this function will interfere 
```

Comment:
_Minor:_ I think this sentence could be reworded for more clarity and providing an example could help.

Reply:
There already exists an example in the init method docs. 
Slight rewording here to ensure the user knows this function will be run before sending some crash report. 

It is possible that the "BeforeSendHandler" is not working as it should, however the description that exists in the documentation describes the current functionality of the BeforeSendHandler

IN PROGRESS